Package request: include-what-you-use

<!--
IMPORTANT: we are not accepting issues about Termux
installations running on Android 5.x-6.x.
-->

**Package description**
IWYU is another static analysis tool for C/C++ languages where it examines every symbol and adds the appropriate `#include` directives for you.
<!--
Describe what this package do and why it should be
available. Make sure that requested package supports
cross-compilation and can run on AArch64 and ARM.
-->

**Link to home page and sources**
1. [Home page:](https://github.com/include-what-you-use/include-what-you-use)
2. [Source code:](https://github.com/include-what-you-use/include-what-you-use)

**Additional information**
Have you compiled or tried to compile the package on device?
* If it didn't work then please provide the error you ran into.
```
 ~/www/cpp/include-what-you-use/build   master  cmake -G "Unix Makefiles" -DIWYU_LLVM_ROOT_PATH=/usr/lib/llvm-6.0 ../                                                                                   127 ↵
-- IWYU: out-of-tree configuration
CMake Error at /data/data/com.termux/files/usr/lib/cmake/llvm/LLVMExports.cmake:1265 (message):
  The imported target "LLVMDemangle" references the file

     "/data/data/com.termux/files/usr/lib/libLLVMDemangle.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/data/data/com.termux/files/usr/lib/cmake/llvm/LLVMExports.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /data/data/com.termux/files/usr/lib/cmake/llvm/LLVMConfig.cmake:243 (include)
  CMakeLists.txt:19 (find_package)


-- Configuring incomplete, errors occurred!
See also "/data/data/com.termux/files/home/www/cpp/include-what-you-use/build/CMakeFiles/CMakeOutput.log".
```

Latest clang, cmake, llvm and libllvm installed.
* If it did work then please share any patches that you had to apply.

Thanks, that got me past the original error but 

```
Setting up libllvm-static (9.0.1-6) ...
 ~/www/cpp/include-what-you-use/build   master  cmake -G "Unix Makefiles" -DIWYU_LLVM_ROOT_PATH=/usr/lib/llvm-6.0 ../                                                                                       ✔
-- IWYU: out-of-tree configuration
CMake Error at /data/data/com.termux/files/usr/lib/cmake/clang/ClangTargets.cmake:327 (message):
  The imported target "clang-import-test" references the file

     "/data/data/com.termux/files/usr/bin/clang-import-test"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/data/data/com.termux/files/usr/lib/cmake/clang/ClangTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /data/data/com.termux/files/usr/lib/cmake/clang/ClangConfig.cmake:18 (include)
  CMakeLists.txt:20 (find_package)


-- Configuring incomplete, errors occurred!
See also "/data/data/com.termux/files/home/www/cpp/include-what-you-use/build/CMakeFiles/CMakeOutput.log".

```

--- a/packages/iwyu/build.sh
+++ b/packages/iwyu/build.sh
@@ -1,0 +1,8 @@
+TERMUX_PKG_HOMEPAGE=https://include-what-you-use.org/
+TERMUX_PKG_DESCRIPTION="A tool to analyze #includes in C and C++ source files"
+TERMUX_PKG_LICENSE=NCSA
+TERMUX_PKG_VERSION=0.14
+TERMUX_PKG_SRCURL=https://github.com/include-what-you-use/include-what-you-use/archive/$TERMUX_PKG_VERSION.tar.gz
+TERMUX_PKG_SHA256=656431b5c66b34ed403ce94c2b2347d3af16e135edc6bc02157a8dfb7d641025
+TERMUX_PKG_DEPENDS='clang, python'
+TERMUX_PKG_BUILD_DEPENDS=libllvm-static

--- a/packages/libllvm/build.sh
+++ b/packages/libllvm/build.sh
@@ -2,6 +2,7 @@ TERMUX_PKG_HOMEPAGE=https://clang.llvm.org/
 TERMUX_PKG_DESCRIPTION="Modular compiler and toolchain technologies library"
 TERMUX_PKG_LICENSE="NCSA"
 TERMUX_PKG_VERSION=10.0.1
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SHA256=(c5d8e30b57cbded7128d78e5e8dad811bff97a8d471896812f57fa99ee82cdf3
 		   f99afc382b88e622c689b6d96cadfa6241ef55dca90e87fc170352e12ddb2b24
 		   591449e0aa623a6318d5ce2371860401653c48bb540982ccdd933992cb88df7a
@@ -14,7 +15,6 @@ TERMUX_PKG_SRCURL=(https://github.com/llvm/llvm-project/releases/download/llvmor
                    https://github.com/llvm/llvm-project/releases/download/llvmorg-$TERMUX_PKG_VERSION/clang-tools-extra-$TERMUX_PKG_VERSION.src.tar.xz)
 TERMUX_PKG_HOSTBUILD=true
 TERMUX_PKG_RM_AFTER_INSTALL="
-bin/clang-import-test
 bin/clang-offload-wrapper
 lib/libgomp.a
 lib/libiomp5.a

--- a/packages/lldb/build.sh
+++ b/packages/lldb/build.sh
@@ -38,11 +38,10 @@ termux_step_host_build() {
 }
 
 termux_step_pre_configure() {
-	# These will be there if libllvm was built from scratch, but not if the pre-built
+	# This will be there if libllvm was built from scratch, but not if the pre-built
 	# package was extracted. Not really needed but the stupid clang CMake config makes
-	# sure they're there.
-	if [ ! -f "$TERMUX_PREFIX/bin/clang-import-test" ]; then
-		touch $TERMUX_PREFIX/bin/clang-import-test
+	# sure it's there.
+	if [ ! -f "$TERMUX_PREFIX/bin/clang-offload-wrapper" ]; then
 		touch $TERMUX_PREFIX/bin/clang-offload-wrapper
 		touch $TERMUX_PKG_BUILDDIR/rm-fake-ci-test
 	fi
@@ -55,6 +54,6 @@ termux_step_make() {
 termux_step_post_make_install() {
 	cp $TERMUX_PKG_BUILDDIR/docs/man/lldb.1 $TERMUX_PREFIX/share/man/man1
 	if [ -f "$TERMUX_PKG_BUILDDIR/rm-fake-ci-test" ]; then
-		rm $TERMUX_PREFIX/bin/clang-{import-test,offload-wrapper}
+		rm $TERMUX_PREFIX/bin/clang-offload-wrapper
 	fi
 }

